src/compiler/parser.ts

```diff
@@ -3073,6 +3074,7 @@ namespace ts {
             switch (operator) {
                 case SyntaxKind.KeyOfKeyword:
                 case SyntaxKind.UniqueKeyword:
+                case SyntaxKind.ReadonlyKeyword:

```

Comment:
Think we need test/handling for .d.ts generation and decorator for this new typeNode kind.